src/systems/filecoin_markets/storage_market/storage_market_actor.go

```diff
@@ -312,6 +319,37 @@ func (a *StorageMarketActorCode_I) GetLastDealExpirationFromDealIDs(rt Runtime,
 	return lastDealExpiration
 }
 
+func (a *StorageMarketActorCode_I) GetUnsealedCIDForDealIDs(rt Runtime, sectorSize util.UVarint, dealIDs []deal.DealID) vmr.InvocOutput {
+	pieceInfos := make([]*sector.PieceInfo_I, len(dealIDs))
+
+	h, st := a.State(rt)
+
+	var totalPieceSize util.UInt
+
+	for index, deal := range st.Deals() {
+		proposal := deal.Proposal()
+		pieceSize := util.UInt(proposal.PieceSize().Total())
+		totalPieceSize += pieceSize
+		pieceInfos[index].PieceCID_ = proposal.PieceCID()
+		pieceInfos[index].Size_ = pieceSize
+	}
+
+	Release(rt, h, st)
+
+	SPS := new(proving.StorageProvingSubsystem_I)
+	ret := SPS.ComputeUnsealedSectorCID(sectorSize, pieceInfos)
+
+	switch ret.Which() {
+	case proving.StorageProvingSubsystem_ComputeUnsealedSectorCID_FunRet_Case_unsealedSectorCID:
+		return rt.ValueReturn(util.Bytes(sector.UnsealedSectorCID(ret.As_unsealedSectorCID())))
```

Comment:
- we dont have access to `StorageProvingSubsystem` from within the VM, only `filproofs`. if there's helpers in there that we need, maybe we should move them out to filproofs, or make another library accessible.
- the reason `StorageProvingSubsystem` doesn't make sense here is that that whole subsystem is an instantiated object/system that only lives in a subset of filecoin nodes, the storge miners.
- in the VM, we need something that is callable in all filecoin nodes, without running any of the storage mining stuff. eg libraries -- calling `filproofs` directly

Reply:
Got it. I wasn't sure what the VM handle on verification should be. Will change.

Reply:
Done.

Reply:
@porcuquine @jbenet @zixuanzh See also https://github.com/filecoin-project/go-filecoin/pull/3636#discussion_r344450703

Reply:
I'll be happy to change this again once we decide on a mechanism for exposing only the intended surface of extra-VM calls to actors.